src/xrpld/app/ledger/detail/InboundLedger.cpp

```diff
@@ -392,7 +392,14 @@ InboundLedger::onTimer(bool wasProgress, ScopedLockType&)
 
     if (!wasProgress)
     {
-        checkLocal();
+        if (checkLocal())
+        {
+            // Done. Something else (probably consensus) built the ledger
+            // locally while waiting for data (or possibly before requesting)
+            assert(isDone());
```

Comment:
please replace with `XRPL_ASSERT` and a short description why; see `CONTRIBUTING.md` for contracts and instrumentation naming guidelines

Reply:
> please replace with `XRPL_ASSERT` and a short description why; see `CONTRIBUTING.md` for contracts and instrumentation naming guidelines

Done